lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/util/HashFactoryTest.java

```diff
@@ -0,0 +1,52 @@
+package uk.gov.di.ipv.cri.drivingpermit.api.util;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.di.ipv.cri.drivingpermit.api.domain.dva.request.DvaPayload;
+
+import java.security.NoSuchAlgorithmException;
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class HashFactoryTest {
+    private HashFactory hashFactory;
+    private DvaPayload testPayload;
+    @Mock private HashFactory.Sha256MessageDigestFactory mockSha256MessageDigestFactory;
+
+    @BeforeEach
+    public void PreEachTestSetup() {
+        testPayload =
+                new DvaPayload(
+                        UUID.randomUUID(),
+                        "Name",
+                        Arrays.asList("Fore"),
+                        LocalDate.of(2023, 8, 1),
+                        LocalDate.of(2023, 8, 1).minusDays(1),
+                        LocalDate.of(2023, 8, 1).plusDays(1),
+                        "12345678",
+                        "AA11 1AA");
+    }
+
+    @Test
+    void hashValidationSuccess() {
```

Comment:
Could we not also get a known hash here and test that the payload passed into this factory returns the expected hash value?

Reply:
I wasn't able to reliably replicate an expected hash value as a UUID is part of the request sent.
So the best I was able to do was use a fixed UUID value in the request data for testing.

Note that because of how UUIDs work, this may affect testing (both locally and in the stub) for some scenarios.